install.sh

```diff
@@ -175,92 +258,46 @@ else
-	instwarn "Terminating installation due to forbidden user"
+	instwarn "Terminating installation due to elevated user permissions"
 	instwarn "The install script should not be run as root"
 	exit 1
 fi
 
-if !($NET_CHECK); then
-
-	# The script will not allow the user to install without internet
-	instwarn "Terminating installation due to the lack of an internet connection"
-	instwarn "The install script needs to be able to connect to GitHub and other sites"
-	exit 1
-fi
-
 
 #===================================================#
 # Repository and Set Up and Main Stack Installation #
 #===================================================#
 
-# Make sure script dependencies are installed on bare bones installations
-instlog "Installing install script dependencies"
-sudo apt-get install -qq wget curl aptitude fakeroot ssh git
-
-# Add software repositories for ROS and Gazebo
-instlog "Adding ROS, Gazebo, and ARM PPAs to software sources"
-sudo sh -c "echo \"deb http://packages.ros.org/ros/ubuntu trusty main\" > /etc/apt/sources.list.d/ros-latest.list"
-sudo sh -c "echo \"deb http://packages.osrfoundation.org/gazebo/ubuntu trusty main\" > /etc/apt/sources.list.d/gazebo-latest.list"
-sudo mkdir -p /etc/apt/preferences.d/
-sudo sh -c "echo 'Package: *\nPin: origin "ppa.launchpad.net"\nPin-Priority: 999' > /etc/apt/preferences.d/arm"
-sudo sh -c "echo \"deb http://ppa.launchpad.net/terry.guo/gcc-arm-embedded/ubuntu trusty main\" > /etc/apt/sources.list.d/gcc-arm-embedded.list"
+# Add software repository for ROS to software sources
+instlog "Adding ROS PPA to software sources"
+sudo sh -c 'echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" > /etc/apt/sources.list.d/ros.list'
+sudo sh -c 'echo "deb-src http://packages.ros.org/ros/ubuntu $(lsb_release -sc) main" >> /etc/apt/sources.list.d/ros.list'
+sudo apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
 
-# Get the GPG signing keys for the above repositories
-wget http://packages.osrfoundation.org/gazebo.key -O - | sudo apt-key add -
-sudo apt-key adv --keyserver hkp://pool.sks-keyservers.net --recv-key 0xB01FA116
-sudo apt-key adv --keyserver hkp://pool.sks-keyservers.net --recv-key 0xA3421AFB
-
-# Add software repository for Git-LFS
+# Add software repository for Git-LFS to software sources
 instlog "Adding the Git-LFS packagecloud repository to software sources"
 curl -s https://packagecloud.io/install/repositories/github/git-lfs/script.deb.sh | sudo bash
 
-# Install ROS and other project dependencies
-instlog "Installing ROS Indigo base packages"
+# Install ROS and a few ROS dependencies
+instlog "Installing ROS $(tr '[:lower:]' '[:upper:]' <<< ${ROS_VERSION:0:1})${ROS_VERSION:1}"
 sudo apt-get update -qq
-sudo apt-get install -qq python-catkin-pkg python-rosdep
 if (env | grep SEMAPHORE | grep --quiet -oe '[^=]*$'); then
-	sudo apt-get install -qq ros-indigo-desktop
+	sudo apt-get install -qq ros-$ROS_VERSION-desktop
```

Comment:
This is a good point, I will install desktop-full on all platforms since we started leaning on it's dependencies so heavily.